test/datahike/test/strategy_test.clj

```diff
@@ -0,0 +1,289 @@
+(ns datahike.test.strategy-test
```

Comment:
Here is an extract of the log output from this test:

```

2023-12-21T12:51:04.405Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:165] - Query related to 1 concept:
2023-12-21T12:51:04.405Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -       select-all: 0.000608
2023-12-21T12:51:04.406Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -    select-simple: 0.000730
2023-12-21T12:51:04.406Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -      expand-once: 0.000916
2023-12-21T12:51:04.406Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -         identity: 0.034990
2023-12-21T12:51:04.491Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:180] - Query related to 2 concepts:
2023-12-21T12:51:04.491Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -      expand-once: 0.000811
2023-12-21T12:51:04.492Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -       select-all: 0.000930
2023-12-21T12:51:04.492Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -    select-simple: 0.036576
2023-12-21T12:51:04.492Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -         identity: 0.044850

....

2023-12-21T12:51:03.904Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:240] - Find all edges between a few small sets of concept types
2023-12-21T12:51:03.905Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -       select-all: 0.001149
2023-12-21T12:51:03.906Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -      expand-once: 0.010314
2023-12-21T12:51:03.906Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -    select-simple: 0.037641
2023-12-21T12:51:03.906Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -         identity: 0.038294

....

2023-12-21T12:51:05.085Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:287] - Find all edges between concept types:
2023-12-21T12:51:05.085Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -         identity: 0.007881
2023-12-21T12:51:05.086Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -      expand-once: 0.009720
2023-12-21T12:51:05.086Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -    select-simple: 0.025386
2023-12-21T12:51:05.086Z jonas-ostlund-jobtech INFO [datahike.test.strategy-test:96] -       select-all: 0.504329

```

Reply:
Given this benchmark `expand-once` would be a much better default.

Reply:
I sort of agree that `expand-once` may be a good default: In none of these examples does it perform worse than `identity` or `select-simple`.

There is one case where we have `select-all: 0.000775` and `expand-once: 0.009825` being more than 10 times slower. So at the moment there doesn't seem to be one-size-fits-all strategy. But the measurements on our API seem to suggest that `expand-once` mostly is just as good as `select-all` or even slightly better.